src/App.tsx

```diff
@@ -14,11 +15,90 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: [],
+  };
+
+  listingOfSelectedGoods = (selectedGoods: string[]) => {
+    if (selectedGoods.length === 1) {
+      return `${selectedGoods[0]} is selected`;
+    }
+
+    const goodsList = selectedGoods.join(', ');
+    const index = goodsList.lastIndexOf(',');
+
+    return `${goodsList.slice(0, index)} and ${goodsList.slice(index + 2)} are selected `;
+  };
+
+  addGoodHandler = (good: string) => {
+    this.setState((state) => ({
+      selectedGoods: state.selectedGoods.concat(good),
+    }));
+  };
+
+  removeGoodHandler = (good: string) => {
+    this.setState((state) => ({
+      selectedGoods: state.selectedGoods.filter(item => item !== good),
+    }));
+  };
+
+  clearHandler = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <h1>
+          Selected good:
+          {selectedGoods.length === 0 ? 'No goods selected' : this.listingOfSelectedGoods(selectedGoods)}
```

Comment:
```suggestion
          {selectedGoods.length === 0 ? 'No goods selected' : this.getCurrentList()}
```
Думаю туда можно параметр не передавать, а доставать его в самом методе из state